scripts/graphql.js

```diff
@@ -1,13 +1,72 @@
 import { getBearerToken } from './security.js';
 import { getAdminConfig } from './site-config.js';
+import { logError } from './scripts.js';
+
+export async function graphqlProductList() {
 
-export async function graphqlCampaignCount() {
   const baseApiUrl = `${await getGraphqlEndpoint()}/graphql/execute.json`;
   const projectId = 'gmo';
-  const queryName = 'campaign-names';
+  const queryName = 'getProductList';
+  //persisted query URLs have to be encoded together with the first semicolon
+  const graphqlEndpoint = `${baseApiUrl}/${projectId}/${queryName}`;
+  const jwtToken = await getBearerToken();
+
+  // Return the fetch promise chain so that it can be awaited outside
+  return fetch(graphqlEndpoint, {
+      method: 'GET',
+      headers: {
+          Authorization: jwtToken,
+      },
+  }).then(response => {
+      if (!response.ok) {
+          throw new Error(`HTTP error! Status: ${response.status}`);
+      }
+      return response.json();
+  }).then(data => {
+      return data; // Make sure to return the data so that the promise resolves with it
+  }).catch(error => {
+      console.error('Error fetching data: ', error);
+      throw error; // Rethrow or handle error as appropriate
+  });
+}
+export async function graphqlStatusList() {
```

Comment:
`graphqlProductList` and `graphqlStatusList` function definition looks same except the query name. Can you combine these methods and have query name passed as parameter?